### PR TITLE
WT-12834 Make perf improvements in setting/getting keys/values

### PR DESCRIPTION
**Pull request by ddanderson**

### CODE REVIEW

File: src/cursor/cur_std.c [hunk 1]
```diff
@@ -411,7 +411,7 @@ __wt_cursor_get_keyv(WT_CURSOR *cursor, uint64_t flags, va_list ap)
     } else {
         /* Fast path some common cases. */
         fmt = cursor->key_format;
-        if (LF_ISSET(WT_CURSOR_RAW_OK) || WT_STREQ(fmt, "u")) {
+        if (WT_STREQ(fmt, "u") || LF_ISSET(WT_CURSOR_RAW_OK)) {
```
Comment (sueloverso) [inline src/cursor/cur_std.c:414]: General comment: Is it worth a comment in these locations where you swapped the order to indicate someone should not change it and/or anything new should get added at the end, etc?

**Review by sueloverso — APPROVED** [review]
LGTM with one possible suggestion.

File: src/cursor/cur_std.c (hunk 1 above)
Comment (ddanderson) [reply src/cursor/cur_std.c:414]: Good idea, I'll add that.

**Review by pmacko86 — APPROVED** [review]
LGTM

File: src/include/cursor_inline.h [hunk 2]
```diff
@@ -52,7 +52,12 @@ __cursor_set_recno(WT_CURSOR_BTREE *cbt, uint64_t v)
 static WT_INLINE int
 __cursor_copy_release(WT_CURSOR *cursor)
 {
-    if (FLD_ISSET(S2C(CUR2S(cursor))->debug_flags, WT_CONN_DEBUG_CURSOR_COPY)) {
+    /*
+     * Make a redundant test first, since it should always fail for high performance situations and
+     * we can exit the function quickly.
+     */
+    if (F_ISSET(cursor, WT_CURSTD_DEBUG_COPY_KEY | WT_CURSTD_DEBUG_COPY_VALUE) &&
+      FLD_ISSET(S2C(CUR2S(cursor))->debug_flags, WT_CONN_DEBUG_CURSOR_COPY)) {
```
Comment (ddanderson) [inline src/include/cursor_inline.h:60]: It turns out that in addition to the `(F_ISSET(cursor, WT_CURSTD_DEBUG_COPY_KEY | WT_CURSTD_DEBUG_COPY_VALUE)` test, we also need to check  `FLD_ISSET(S2C(CUR2S(cursor))->debug_flags, WT_CONN_DEBUG_CURSOR_COPY)` like the original code.  Or else wt2323_join fails under ASAN.

I think that DEBUG_COPY_KEY or DEBUG_COPY_VALUE can get set even if WT_CONN_DEBUG_CURSOR_COPY is not set.   That happens at the end of __wt_cursor_set_valuev/keyv.

**Comment by ddanderson** [comment]
@pmacko86 and @sueloverso can you please review again?  See my comment in src/include/cursor_inline.h where I needed to make a slight adjustment.

**Review by pmacko86 — APPROVED** [review]
LGTM

**Pull request merged**